### ISSUE TITLE
Using confint.gnm() inside a function

### ISSUE THREAD

**Issue by dwoll** [hturner/gnm#11]
I believe this is related to issue 1. Consider

```
library(gnm)
f <- function(d) {
    fit <- gnm(Freq ~ vote, eliminate=class, family=poisson, data=d)
    confint(fit)
}
f(as.data.frame(cautres))
```

For me, this fails because `d` could not be found - presumably because some function downstream from `update.gnm()` does not search in the correct environment. I'm sorry I'm not able to locate the exact source of the error, but I don't fully understand what's going on with environments in the following evaluated calls to `gnm()`, `gnmTerms()` etc.

Thanks for quickly fixing the previous issue and for your ongoing efforts!

**Comment by kauedesousa** [comment]
Hi @dwoll  

Seems that it works well when you avoid the argument "eliminate". Do you have any especial reason for using this argument?

library(gnm)
f <- function(d) {
  fit <- gnm(Freq ~ vote, family = poisson, data = d)
  confint(fit)
}
f(as.data.frame(cautres))

**Comment by dwoll** [comment]
@kauedesousa Yes, I need the conditional Poisson regression because I have to stratify for a lot of factors that are, by themselves, uninteresting. However, even without the `eliminate` option, I still get an error from `terms.formula()`.

**Comment by hturner** [comment]
Okay, this is going to be harder to get right. This is a workaround for your example:

    library(gnm)
    f <- function(d) {
        fit <- do.call("gnm", list(Freq ~ vote, eliminate=quote(class), family=poisson, data=d))
        confint(fit)
    }
    f(as.data.frame(cautres))

Hopefully you can adapt that for your application. If you need to specify the eliminated variable you can do it as follows:

    f <- function(d, x) {
        fit <- do.call("gnm", list(Freq ~ vote, eliminate=substitute(x), family=poisson, data=d))
        confint(fit)
    }
    f(as.data.frame(cautres), class)

I think it will be a little while before I can implement a proper fix for this.
